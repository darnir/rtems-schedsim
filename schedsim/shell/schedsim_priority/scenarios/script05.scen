echo "*** TEST 05 ***"
rtems_init
echo Create 0x0a010001 
task_create TA1 1
task_create TA2 2
task_create TA3 3
semaphore_create SEM1
semaphore_obtain SEM1 0
semaphore_obtain SEM1 0
semaphore_obtain SEM1 0
semaphore_flush SEM1
echo "*** Now back to TA1 to do it differently"
semaphore_obtain SEM1 0
semaphore_obtain SEM1 0
semaphore_release SEM1
# as TA1
semaphore_obtain SEM1 0
semaphore_delete SEM1
# should preempt
echo "*** END OF TEST 05 ***"
exit
# We will not get here
