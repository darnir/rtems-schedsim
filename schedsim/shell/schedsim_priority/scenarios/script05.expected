*** TEST 05 ***
  CPU 0: Thread Heir: 0x09010001 priority=255
  CPU 0: Thread Executing: 0x09010001 priority=255
Create 0x0a010001
Creating task with:  Preemption: yes Timeslicing: no
Task (TA1) created: id=0x0a010001, priority=1
Task (TA1) starting: id=0x0a010001, priority=1
  CPU 0: Thread Heir: 0x0a010001 priority=1
  CPU 0: Thread Executing: 0x0a010001 priority=1
Creating task with:  Preemption: yes Timeslicing: no
Task (TA2) created: id=0x0a010002, priority=2
Task (TA2) starting: id=0x0a010002, priority=2
Creating task with:  Preemption: yes Timeslicing: no
Task (TA3) created: id=0x0a010003, priority=3
Task (TA3) starting: id=0x0a010003, priority=3
Semaphore (SEM1) created: id=0x1a010001
Obtain semaphore (0x1a010001) with timeout 0
  CPU 0: Thread Heir: 0x0a010002 priority=2
  CPU 0: Thread Executing: 0x0a010002 priority=2
Obtain semaphore (0x1a010001) with timeout 0
  CPU 0: Thread Heir: 0x0a010003 priority=3
  CPU 0: Thread Executing: 0x0a010003 priority=3
Obtain semaphore (0x1a010001) with timeout 0
  CPU 0: Thread Heir: 0x09010001 priority=255
  CPU 0: Thread Executing: 0x09010001 priority=255
Flushing semaphore (0x1a010001)
  CPU 0: Thread Heir: 0x0a010001 priority=1
  CPU 0: Thread Executing: 0x0a010001 priority=1
*** Now back to TA1 to do it differently
Obtain semaphore (0x1a010001) with timeout 0
  CPU 0: Thread Heir: 0x0a010002 priority=2
  CPU 0: Thread Executing: 0x0a010002 priority=2
Obtain semaphore (0x1a010001) with timeout 0
  CPU 0: Thread Heir: 0x0a010003 priority=3
  CPU 0: Thread Executing: 0x0a010003 priority=3
Releasing semaphore (0x1a010001)
  CPU 0: Thread Heir: 0x0a010001 priority=1
  CPU 0: Thread Executing: 0x0a010001 priority=1
Obtain semaphore (0x1a010001) with timeout 0
  CPU 0: Thread Heir: 0x0a010003 priority=3
  CPU 0: Thread Executing: 0x0a010003 priority=3
Deleting semaphore (0x1a010001)
  CPU 0: Thread Heir: 0x0a010001 priority=1
  CPU 0: Thread Executing: 0x0a010001 priority=1
*** END OF TEST 05 ***
